src/gif.rs

```diff
@@ -201,12 +182,17 @@ impl<R: Read> Iterator for GifFrameIterator<R> {
         }
 
         // create the image buffer from the raw frame
-        let mut image_buffer = match ImageBuffer::from_raw(self.width, self.height, vec) {
-            Some(buffer) => buffer,
-            None => return Some(Err(ImageError::UnsupportedError(
-                "Unknown error occured while reading gif frame".into()
-            ))),
-        };
+        let image_buffer_raw = ImageBuffer::from_raw(f_width, f_height, vec).unwrap();
```

Comment:
Similar to the other comment, the previous code already tested for the error case even though its wording was very imprecise.